expdj/apps/experiments/models.py

```diff
@@ -176,6 +202,61 @@ def save(self, *args, **kwargs):
         assign_perm('del_battery', self.owner, self)
         assign_perm('edit_battery', self.owner, self)
 
+    def check_battery_dependencies(self, worker_id):
+        '''
+        check_battery_dependencies looks up all of a workers completed 
+        experiments in a result object and places them in a dictionary 
+        organized by assignment_id. Each of these buckets of results is
+        iterated through to check that every experiment in a battery has
+        been completed. In this way a list of batteries that a worker has 
+        completed is built. This list is then compared to the lists of 
+        required and restriected batteries to determine if the worker is 
+        elidgble to attempt the current(self) battery.
+        '''
+        worker_results = turk.models.Result.objects.filter(
```

Comment:
I like this function, but I don't think it should be part of the Battery Model. It integrates functionality from the Turk and Experiments module, and seems more fit for a tasks.py function. Depending on the user of the experiment factory, a battery does not necessarily need to have an understanding of a worker status, and I'd go as far to say that an ideal application would have subject management separately from the experiments. I think this should be moved to tasks.py.